macros/macros.go

```diff
@@ -0,0 +1,48 @@
+package macros
+
+import (
+	"regexp"
+	"strings"
+	"text/template"
+)
+
+// EndpointTemplateParams specifies params for an endpoint template
+type EndpointTemplateParams struct {
+	Host        string
+	PublisherID int
+}
+
+// UserSyncTemplateParams specifies params for an user sync URL template
+type UserSyncTemplateParams struct {
+	GDPR        string
+	GDPRConsent string
+}
+
+// MacroRegExp is the regexp to check if a string contains a macro or not
+const MacroRegExp string = `\{\{\.(\w+)\}\}`
+
+// BuildTemplate builds a text template for a given string
+func BuildTemplate(templateStr string) (*template.Template, error) {
```

Comment:
Just food for thought... compare these two equivalent lines of code from a caller:

```
t, err := macros.BuildTemplate(s)
t, err := template.New("aTemplate").Parse(s)
```

Are the keystrokes it saves you worth the cost of testing + maintaining it?

Personally I don't think so, but... your call in the end.

Reply:
Yeah, I agree. This is more of a leftover from another approach that I was trying. Will remove that method. Thanks for pointing it out!